### ISSUE TITLE
intercept can't catch requests for cached by browser file.

### ISSUE THREAD

**Issue by SchDen** [cypress-io/cypress#16766]
### Current behavior

I use `cy.intercent` to handle download static file (`stl`) and if I have only one test with intercept this file, it works good, but if I have two+ tests with the same intercept - cypress can't catch this request so I get the error "No request ever occurred".

Even if I use one test, it will work the first time, but if I click on the relaunch button (hotkey "R"), the test will fail.

I researched and found interesting facts: 
- if open the chrome developer tools and disable cache in the "Network" tab, it will always work correctly
- It works well with `.json` files because a browser doesn't use catch for this type

In summary, I think cypress has a problem with browser cache.

### Desired behavior

`cy.intercept` catch requests all time

### Test code to reproduce

I create test env in this repository: https://github.com/SchDen/cypress-test-tiny

```
- `npm i`
- `npm run dev`
- `npm run cypress:open`
- Just run test
```

Important: as soon test fail - open chrome developer tools and disable cache - after that, relaunch
<img width="867" alt="Screenshot 2021-06-02 at 01 56 41" src="https://user-images.githubusercontent.com/1484977/120400062-de9e2d00-c345-11eb-86be-816fedebb68a.png">

### Versions
Current version: v7.4.0
Worked well on: v4.12.0

**Comment by SchDen** [comment]
In this branch cypress v4.12.1, and it works well: https://github.com/SchDen/cypress-test-tiny/tree/cypress-v4.12.0

**Comment by SchDen** [comment]
Small hack for fixing it there, but the problem with intercept still actual

https://stackoverflow.com/a/67858001/2660839

**Comment by flotwig** [comment]
@SchDen This is a caveat of how `cy.intercept()` works - if a request never hits the network layer, it can not be intercepted by `cy.intercept()`. Requests served from cache can happen without sending a request over HTTP.

I've opened a PR to the docs repo with more clarification on this, as well as workarounds: https://github.com/cypress-io/cypress-documentation/pull/4042

Closing this issue since at this time there is not a path to change this behavior.
